app/src/main/java/org/p2p/wallet/home/ui/main/empty/BigBannerViewHolder.kt

```diff
@@ -0,0 +1,47 @@
+package org.p2p.wallet.home.ui.main.empty
+
+import android.graphics.drawable.ColorDrawable
+import android.graphics.drawable.Drawable
+import android.graphics.drawable.GradientDrawable
+import android.graphics.drawable.ShapeDrawable
+import android.view.ViewGroup
+import androidx.annotation.ColorRes
+import androidx.recyclerview.widget.RecyclerView
+import org.p2p.wallet.databinding.ItemBigBannerBinding
+import org.p2p.wallet.home.model.HomeBannerItem
+import org.p2p.wallet.utils.viewbinding.getColor
+import org.p2p.wallet.utils.viewbinding.inflateViewBinding
+
+class BigBannerViewHolder(
+    parent: ViewGroup,
+    private val binding: ItemBigBannerBinding = parent.inflateViewBinding(attachToRoot = false),
+    private val onBannerButtonClicked: (buttonId: Int) -> Unit,
+) : RecyclerView.ViewHolder(binding.root) {
+
+    fun onBind(item: HomeBannerItem) = with(binding) {
+        textViewBannerTitle.setText(item.titleTextId)
+        textViewBannerSubtitle.setText(item.subtitleTextId)
+
+        imageViewBanner.setImageResource(item.drawableRes)
+
+        buttonBanner.apply {
+            setText(item.buttonTextId)
+            setOnClickListener { onBannerButtonClicked(item.id) }
+        }
+        setBackground(item.backgroundColorRes)
+    }
+
+    private fun ItemBigBannerBinding.setBackground(@ColorRes backgroundColorRes: Int) {
+        when (val background: Drawable = viewBannerBackground.background) {
+            is ShapeDrawable -> {
+                background.paint.color = getColor(backgroundColorRes)
+            }
+            is GradientDrawable -> {
+                background.setColor(getColor(backgroundColorRes))
+            }
+            is ColorDrawable -> {
+                background.color = getColor(backgroundColorRes)
+            }
```

Comment:
Why do you need this check if background is ShapeDrawable?

Reply:
depending on background